terraform/eval_variable.go

```diff
@@ -175,14 +174,22 @@ func (n *EvalVariableBlock) setUnknownVariableValueForPath(path string) error {
 	var current interface{} = n.VariableValues[pathComponents[0]]
 	for i := 1; i < len(pathComponents); i++ {
 		switch current.(type) {
```

Comment:
We could define `switch tCurrent := current.(type) {` here, and get rid of the redundant assertions below.